src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -216,7 +216,41 @@ public void DoDamage(DamageType damageType, Fix64 amount, DeathType deathType, T
 
         public int Supply { get; set; }
 
-        public List<string> HiddenSubObjects;
+        private Dictionary<string, bool> _hiddenSubObjects;
+        private Dictionary<string, bool> _shownSubObjects;
+
+        public void HideSubObject(string subObject, bool permanent = false)
+        {
+            if (!_hiddenSubObjects.ContainsKey(subObject))
+            {
+                _hiddenSubObjects.Add(subObject, permanent);
```

Comment:
With this logic, if there’s an existing object, it won’t have it’s “permanent” value updated with the new value passed in here. Is that intended?

Reply:
indeed, should be fixed now